### PR TITLE
Add SwitchCaseOnNewline rule

### PR DESCRIPTION
**Pull request by marcelofabri**
This fixes https://github.com/realm/SwiftLint/issues/681

I actually haven't checked if the code is inside a `switch`, ~~so this is currently triggering a false positive: `if let case .someEnum(value) = aFunction([key: 2]) {`.~~

~~I thought about checking if `case` is the first keyword in the line instead of checking if it's inside a `switch`, as I think the later is trickier. Thoughts?~~

Update: I added a check to see if `case` is the first keyword in the line

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/SwitchCaseOnNewlineRule.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        triggeringExamples: [
+            "case 1: return true",
+            "case let value: return true",
+            "default: return true",
+            "case \"a string\": return false"
+        ]
+    )
+
+    public func validateFile(file: File) -> [StyleViolation] {
+        let pattern = "(case[^\n]*|default):[^\\S\n]*[^\n]+"
```
Comment (norio-nomura) [inline Source/SwiftLintFramework/Rules/SwitchCaseOnNewlineRule.swift]: Sorry, I forgot to wrote in last review.
I think trailing `+` won't be needed.

**Comment by marcelofabri** [comment]
Thanks for reviewing this! I've addressed on the latest commit 👍

**Comment by norio-nomura** [comment]
Thanks! 😄

**Comment by jpsim** [comment]
Nice work! 🎉

**Pull request merged**